lua/entities/gmod_tardis/modules/sh_redecorate.lua

```diff
@@ -0,0 +1,175 @@
+	end)
+
+	function ENT:Redecorate()
+
+		local ply = self:GetCreator()
+
+		self:SetData("repairing", false, true)
+		self:ChangeHealth(TARDIS:GetSetting("health-max"))
+		self:SetPower(true)
+
+		-- save tardis state
+		local ext_saved_data = {}
+		local int_saved_data = {}
+		for k,v in ipairs(ext_saved_data_names) do
+			ext_saved_data[v] = self:GetData(v)
+		end
+		if IsValid(self.interior) then
+			for k,v in ipairs(int_saved_data_names) do
+				int_saved_data[v] = self.interior:GetData(v)
+			end
+		end
+
+		local child = TARDIS:SpawnTARDIS(ply, {
+			pos = self:GetPos(),
+			metadataID = self:GetData("redecorate-interior") or "default",
+			redecorate_parent = self,
+			ext_data = ext_saved_data,
+			int_data = int_saved_data,
+		})
+
+		if not IsValid(child) then
+			error("Redecoration failed: failed to spawn the replacement")
+		end
+		return IsValid(child) -- if gone wrong, finish repair
+	end
+
+	ENT:AddHook("StopDemat", "redecorate_remove_parent", function(self)
+		if self:GetData("redecorate") then
+			self:GetData("redecorate_child"):SetData("redecorate_parent", nil, true)
+			self:Remove()
+		end
+	end)
+
+	ENT:AddHook("MatStart", "redecorate_sync", function(self)
+		local parent = self:GetData("redecorate_parent")
+		if not parent then return end
+
+		self:SetCollisionGroup(COLLISION_GROUP_WORLD)
+		self:SetPos(parent:GetPos())
+		self:SetAngles(parent:GetAngles())
+		parent:SetParent(self)
+	end)
+
+	ENT:AddHook("CustomData", "redecorate_child", function(self, customdata)
+		local parent = customdata.redecorate_parent
+		if parent then
+			self:SetData("redecorate_parent", parent)
+			parent:SetData("redecorate_child", self)
+
+			self:SetPos(parent:GetPos())
+			self:SetAngles(parent:GetAngles())
+
+			self:SetData("redecorate_parent_int_data", customdata.int_data, true)
+			self:SetData("redecorate_parent_ext_data", customdata.ext_data, true)
+		end
+	end)
+
+	ENT:AddHook("Initialize", "redecorate_child", function(self)
+		local parent = self:GetData("redecorate_parent")
+		if not parent then return end
+
+		local phys = self:GetPhysicsObject()
+		phys:Sleep()
+
+		self:SetCollisionGroup(COLLISION_GROUP_IN_VEHICLE)
+	end)
+
+	ENT:AddHook("PostInitialize", "redecorate_child", function(self)
+		local parent = self:GetData("redecorate_parent")
+		if not parent then return end
+
+		self:ForceDematState()
+
+		local ext_saved_data = self:GetData("redecorate_parent_ext_data")
+		local int_saved_data = self:GetData("redecorate_parent_int_data")
+		local phys = self:GetPhysicsObject()
+
+		apply_saved_data(self, ext_saved_data, ext_saved_data_names)
+		apply_saved_data(self.interior, int_saved_data, ext_saved_data_names)
+
+		self:SetData("redecorate_parent_ext_data", nil, true)
+		self:SetData("redecorate_parent_int_data", nil, true)
+
+		constraint.RemoveAll(parent) -- drop everything attached
+		parent:SetFastRemat(true)
+		parent:ForcePlayerDrop()
```

Comment:
I've added safeguards to prevent that one in a25844d5f13117fd639e2c45d1f69c777d3d8622
Now, when the parent is dropped, it cannot be picked up or moved again (COLLISION_GROUP_IN_VEHICLE)
+ it won't move by itself due to physlock